frame/curve-amm/src/lib.rs

```diff
@@ -600,9 +713,66 @@ pub mod pallet {
 				sent_amount: dx,
 				received_token_index: j,
 				received_amount: dy,
+				fee,
 			});
 			Ok(())
 		}
+
+		fn withdraw_admin_fees(
+			who: &Self::AccountId,
+			pool_id: PoolId,
+			admin_fee_account: &Self::AccountId,
+		) -> Result<(), DispatchError> {
+			let admin_fees =
+				Pools::<T>::try_mutate(pool_id, |maybe_pool| -> Result<_, DispatchError> {
+					let pool = maybe_pool.as_mut().ok_or(Error::<T>::PoolNotFound)?;
+					let n_coins = pool.assets.len();
+
+					ensure!(n_coins == pool.balances.len(), Error::<T>::InconsistentStorage);
+
+					let total_balances = &pool.total_balances;
+					let balances = &pool.balances;
+
+					let admin_fees = total_balances
+						.into_iter()
+						.zip(balances)
+						.map(|(tb, b)| {
+							let admin_fee = tb.checked_sub(b).ok_or(Error::<T>::Math)?;
+
+							Ok(admin_fee)
+						})
+						.collect::<Result<Vec<Self::Balance>, DispatchError>>()?;
+
+					let assets = pool.assets.clone();
+
+					for (asset, amount) in assets.into_iter().zip(admin_fees.iter().copied()) {
+						T::LpToken::transfer(
+							asset,
+							&Self::account_id(&pool_id),
+							admin_fee_account,
+							amount,
+							true,
+						)?;
+					}
+
+					for i in 0..n_coins {
+						pool.total_balances[i] = pool.total_balances[i]
+							.checked_sub(&admin_fees[i])
+							.ok_or(Error::<T>::Math)?;
+					}
```

Comment:
Can `Verify first, write last` (https://docs.substrate.io/v3/runtime/storage/#verify-first-write-last) violate here? i.e. there will be a write to the database in the first, then will be an error